src/Microsoft.Azure.WebJobs.Logging/Internal/LogWriter.cs

```diff
@@ -329,6 +330,35 @@ private static void Increment(FunctionInstanceLogItem item, TimelineAggregateEnt
 
 
             await Task.WhenAll(t);
-        }    
+        }
+
+        // Collection where adding in the same RowKey replaces a previous entry with that keyl. 
+        // This is single-threaded. Caller must lock. 
+        private class RowKeyCollection<T> where T : TableEntity
```

Comment:
This type confuses me. It's not really a collection of row keys, its a collection of TableEntities. Also, it doesn't really seem to add much behavior above Dictionary itself. I.e., you could just use a Dictionary instead of this type and use the indexer for assignment rather than Add.